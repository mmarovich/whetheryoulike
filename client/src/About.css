.about-container {
    padding: 30px 20px;
    text-align: center;
}

.go-back {
    margin-top: 50px;
}