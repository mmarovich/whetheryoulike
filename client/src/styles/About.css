.about-container {
    padding: 30px 20px;
    text-align: left;
}

.about-container > div {
    padding: 15px;
}

.go-back {
    margin-top: 50px;
}