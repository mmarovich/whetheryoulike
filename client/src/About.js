import React, { Component } from 'react'
import { Link } from 'react-router-dom'

import './About.css'

export default class About extends Component {
    render() {
        return(
            <div className="about-container">
                <div className="purpose">
                    <h3>Purpose</h3>
                    <p>WhetherYouLike is a mobile app prototype that creates a customized
                        background for your mobile device.  You can switch on various information,
                        such as time, date, weather, and webcams, as well as change color themes
                        for both the background and elements.  After that, you have the ability
                        to drag those elements where ever you would like them to be on-screen.

                        This is more versatile than a typical widget, where you drag the widget around
                        within a 3 by 6 grid, effecting the position of your icons.  This sets the
                        background of your mobile device so that your icons and other widgets remain
                        where you want them.
                    </p>
                </div>
                <div className="goals">
                    <h3>Goals</h3>
                    <p>This project is going to be converted to the mobile app language of android
                        (React Native, and perhaps later on, Kotlin).  I can also see more elements and
                        options being added over time. As an example, temperature highs and lows for the
                        day, or record highs and lows from the past.  I'd also like to add a feature where
                        instead of a solid background, the user has the choice to upload a picture of their
                        choice.
                    </p>
                </div>
                <p className="go-back"><Link to="/settings">Go back</Link></p>
            </div>
        )
    }
}