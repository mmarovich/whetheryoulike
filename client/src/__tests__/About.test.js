import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import About from '../About'

describe('About component', () => {
    it('renders About', () => {

        const tree = renderer.create(
            <MemoryRouter>
                <About />
            </MemoryRouter>).toJSON()
        expect(tree).toMatchSnapshot();
    })
})