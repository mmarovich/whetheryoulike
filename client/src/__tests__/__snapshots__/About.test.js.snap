// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders About 1`] = `
<div
  className="about-container"
>
  <div
    className="purpose"
  >
    <h3>
      Purpose
    </h3>
    <p>
      WhetherYouLike is a mobile app prototype that creates a customized background for your mobile device.  You can switch on various information, such as time, date, weather, and webcams, as well as change color themes for both the background and elements.  After that, you have the ability to drag those elements where ever you would like them to be on-screen. This is more versatile than a typical widget, where you drag the widget around within a 3 by 6 grid, effecting the position of your icons.  This sets the background of your mobile device so that your icons and other widgets remain where you want them.
    </p>
  </div>
  <div
    className="goals"
  >
    <h3>
      Goals
    </h3>
    <p>
      This project is going to be converted to the mobile app language of android (React Native, and perhaps later on, Kotlin).  I can also see more elements and options being added over time. As an example, temperature highs and lows for the day, or record highs and lows from the past.  I'd also like to add a feature where instead of a solid background, the user has the choice to upload a picture of their choice.
    </p>
  </div>
  <p
    className="go-back"
  >
    <a
      href="/settings"
      onClick={[Function]}
    >
      Go back
    </a>
  </p>
</div>
`;
